lib/base-compiler.ts

```diff
@@ -1086,9 +1092,9 @@ export class BaseCompiler implements ICompiler {
         let staticLibLinks: string[] = [];
 
         if (filters.binary) {
-            libLinks = this.getSharedLibraryLinks(libraries) || [];
+            libLinks = (this.getSharedLibraryLinks(libraries).filter(l => l) as string[]) || [];
```

Comment:
There are a lot of `(string | false)[]` shenanigans from library related stuff. It might make sense to get rid of those closer to the root of the problem.